### ISSUE TITLE
Use native enums if the database supports them

### ISSUE THREAD

**Issue by thedrow** [5monkeys/django-enumfield#18]
All major databases that Django support already supports enums at the database level.
* Postgres supports [enums](http://postgresguide.com/sexy/enums.html)
* MySQL supports [enums](http://dev.mysql.com/doc/refman/5.7/en/enum.html)
* Oracle supports [enums](https://docs.oracle.com/cd/E17952_01/refman-5.6-en/enum.html)

You can also implement [enums in SQLite using CHECK](http://stackoverflow.com/questions/5299267/how-to-create-enum-type-in-sqlite#answer-17203007).

Is there a good way to hook into south and django migrations framework in order to make this field use their native database type?

**Comment by hannseman** [comment]
Good idea, it would be interesting to try it out. It could be as simple as overriding `db_type` on the `EnumField`-class. 

https://docs.djangoproject.com/en/dev/howto/custom-model-fields/#custom-database-types

**Comment by thedrow** [comment]
In postgres you have to run CREATE TYPE first so a migration is required.
On the other databases it should be a matter of overrriding db_type.

**Comment by AncientSwordRage** [comment]
Any progress on this?

**Comment by thedrow** [comment]
I'm gonna try to implement it when I'm in London.
The next Django sprint will take place there.

**Comment by kjagiello** [comment]
"2. Changing the member list of ENUM columns is very expensive."
 http://komlenic.com/244/8-reasons-why-mysqls-enum-data-type-is-evil/

**Comment by kaidokert** [comment]
I'm looking at a different path, which would have enum values in a classic lookup table with foreign key instead. The table has auto_id, enum_value, enum_label , and i'm trying to come up with a reasonably safe sync method that updates existing databases with new model code from the enum classes as needed.

django-enumfield is cool as its implemented, but i really want to enforce referential integrity at the db level, and also have my database make sense to someone who has never looked at django/python at all.

If someone has done this already, i'd appreciate a pointer.

**Comment by pjdelport** [comment]
Upstream Django issue: [#24342: Add EnumField model/form fields](https://code.djangoproject.com/ticket/24342)

Related implementation: https://github.com/ashleywaite/django-more/tree/master/django_enum

**Comment by HaddadJoe** [comment]
is this a stale issue? looks like the issue is dormant on the upstream django tickets board. it would be nice if this library can support it. 

Happy to open a PR if it would be considered
